src/operator/every.ts

```diff
@@ -8,38 +8,41 @@ import {Subscriber} from '../Subscriber';
 import {tryCatch} from '../util/tryCatch';
 import {errorObject} from '../util/errorObject';
 
+export type Predicate<T> = (value: T, index: number, source: Observable<T>) => boolean;
+
 /**
  * Returns an Observable that emits whether or not every item of the source satisfies the condition specified.
  * @param {function} predicate a function for determining if an item meets a specified condition.
  * @param {any} [thisArg] optional object to use for `this` in the callback
  * @returns {Observable} an Observable of booleans that determines if all items of the source Observable meet the condition specified.
  */
-export function every<T>(predicate: (value: T, index: number, source: Observable<T>) => boolean,
-                         thisArg?: any): Observable<boolean> {
-  const source = this;
-  if (source._isScalar) {
-    const result: boolean = tryCatch(predicate).call(thisArg || this, source.value, 0, source);
+export function every<T>(predicate: Predicate<T>, thisArg?: any): Observable<boolean> {
+    
+  let _this: Observable<T> = this;
+  if (_this._isScalar) {
+    let scalar: ScalarObservable<T> = <any>_this;
```

Comment:
I'll also point out that the extra emits have an effect proportional to the number of calls to `every` in the program, not the number of items that pass through the `every` operator. An large sized program will have a thousand to two thousand invocations (generously) of the operator functions, so an extra local variable per shouldn't have any performance impact.

All this aside, I'll be removing the `_this` aliases for this PR and will open a separate issue to discuss the pros/cons of it.